// Copyright 2019 The Google AI Language Team Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto2";

package language.tapas;

import "tapas/protos/interaction.proto";

message AnnotatedText {
  extend language.tapas.Question {
    optional AnnotatedText annotated_question_ext = 304163798;
  }
  extend language.tapas.Cell {
    optional AnnotatedText annotated_cell_ext = 304163798;
  }
  repeated Annotation annotations = 1;
}

message AnnotationDescription {
  extend language.tapas.Interaction {
    optional AnnotationDescription annotation_descriptions_ext = 319400515;
  }
  // For each entity that appears in the interaction, the map has a textual
  // description of the entity, like the first section of its Wikipedia page.
  map<string, string> descriptions = 1;
}

message Annotation {
  // Indices refer to 'original_text' of a question or 'text' of a cell.
  // Inclusive begin byte.
  optional int64 begin_byte_index = 1;
  // Exclusive end byte.
  optional int64 end_byte_index = 2;
  // An identifier for example a Wikipedia URL.
  optional string identifier = 3;
}
